presto-product-tests-utils/pom.xml

```diff
@@ -0,0 +1,79 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <artifactId>presto-root</artifactId>
+        <groupId>com.facebook.presto</groupId>
+        <version>0.116-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>presto-product-tests-utils</artifactId>
+    <name>presto-product-tests-utils</name>
+
+    <properties>
+        <air.main.basedir>${project.parent.basedir}</air.main.basedir>
+    </properties>
+
+    <dependencies>
+        <dependency>
+            <groupId>com.facebook.presto</groupId>
+            <artifactId>presto-main</artifactId>
+            <exclusions>
+                <exclusion>
+                    <groupId>org.slf4j</groupId>
+                    <artifactId>slf4j-jdk14</artifactId>
```

Comment:
> >  We store test name in MDC map and then use it in logging pattern. We could log thread id and and then do the matching by it, but it is less user friendly. So I would opt for keeping logback as logging backend here. This should not be a problem as this is not part of the product.
Can you elaborate on how logging is used in the testing framework?

> In my experience, relying on thread-locals for carrying state (e.g., logback's MDC) is fragile -- once you make your code multi-threaded (executors, thread pools, etc), the model stops working gets much harder to deal with.

I agree that generally one should avoid thread-locals. Yet we decided that in case of tempto, despite the fact that limits possibility of using Executors in tests code, they do more good than harm. We use them for two things.

1. We use SLF4J MDC for storing executed test name. It is set and cleaned by `TestInitializationListener` which is TestNG listener coordinating test execution. By having testcases name in MDC we can run tests in parallel (which actually we do not do right now ;) ) and still know which log line is coming from which testcase.

2. Each running test has reference of its `TestContext` on thread local variable. With a bit of simplification `TestContext` is wrapper around guice's `Injector` holding all the external dependencies available to test. As you noted this is somewhat fragile, and requires test writers to follow specific guidelines, in terms of using threads in test-code (we provide `TestContextDsl` class for simplifying that). On the other hand it allows us to expose static helper functions like `QueryExecutor#query` and not pass actual dependencies everywhere explicitly. IMO This makes test writing easier. And tests more readable.